torch/csrc/distributed/rpc/PythonRpcHandler.cpp

```diff
@@ -0,0 +1,89 @@
+#include <torch/csrc/distributed/rpc/PythonRpcHandler.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+PyObject* PythonRpcHandler::module_ = nullptr;
+PyObject* PythonRpcHandler::runUDFFunction_ = nullptr;
+PyObject* PythonRpcHandler::loadResultFunction_ = nullptr;
+
+void PythonRpcHandler::init() {
+  Py_Initialize();
```

Comment:
I think we need to discuss more about this line.

It looks to me that our user starts with launching a Python interpreter (a C program) to interpret a piece of Python code. This Python interpreter, loads a pybind11 module, `torch.distributed`, that bundles several C++ classes.
We use this special binding module, backed by C++ classes, to instantiate a `RpcAgent` C++ instance, and assigned it to a variable `torch.distributed.rpc._agent`.
While, in the constructor of `RpcAgent`, `PythonRpcHandler::init()` is called and creates a **second** Python interpreter ?? (If I understand the [doc](https://docs.python.org/3/c-api/init.html#c.Py_Initialize) correctly)
This **second** Python interpreter imports the module, `torch.distributed.internal_rpc_utils`, and use the utilities in it. 
Although it can successfully run the function sent by the client and return correct results back to the client.
It's impossible for the client to change the global state in the server, which is hold by the **first** Python interpreter.

---
There is a requirement in our RRef prototype implementation that we need to allow an RPC client (an RRef user), to dictate the RPC server (RRef owner) to change it's global state. 
The state is stored in a module's global variable, which lives in the original Python Interpreter.

For example, 
1) A client could ask the server to fetch an RRef instance that matches the `ref_id` in the message, from the server's global RRef registry.
2) A client could send termination signal message to a server, the server, on receiving the signal, should set it's status to `termination_ongoing=True`, so that it will delete the remote references it holds and notifies the owners about reducing ref counts before it goes away.